src/ReactIntl.re

```diff
@@ -85,34 +102,97 @@ type translation = {
   "message": Js.nullable(string),
 };
 
+module Part = {
+  type t;
+
+  [@bs.get] external type_: t => string = "type";
+  [@bs.get] external value: t => string = "value";
+};
```

Comment:
With objects as records in BS 7, I would model this as a record.

Reply:
Agreed, done.

We prolly should update `message` & `translation` types as well. But the former would require changes in the extractor AFAICT.

Reply:
Right, that's a good idea! I can look into updating the extractor sometime next week.

This should allow for a better API, too. Instead of

```reason
let msg = [@intl.messages] {
                             "helloWorld": {
                               "id": "MyComponent.helloWorld",
                               "defaultMessage": "Hello, world!",
                             },
                           };
```

(strange indentation by refmt...) we could e.g. have

```reason
[@intl.messages]
module Messages = {
  open ReactIntl;

  let notSpecified = {id: "MyComponent.helloWorld", defaultMessage: "Hello, world!"};
};
```

😃

Reply:
Sounds perfect.

AFAICT we can even drop sub-module and define annotation on the module level:

```reason
[@intl.messages];

open ReactIntl;

let helloWorld = {id: "MyComponent.helloWorld", defaultMessage: "Hello, world!"};
```

Reply:
Yes, if you have a "global message module" (containing all the application's messages), that makes sense. I can look into adding this usage pattern to the extractor, too.

In my apps, I prefer having smaller submodules that contain just the messages used by their parent module.

Reply:
Right, and this should still work for sub-modules:

```reason
module Msg = {
  [@intl.messages];

  open ReactIntl;

  let helloWorld = {id: "MyComponent.helloWorld", defaultMessage: "Hello, world!"};
};
```

Reply:
@alexfedoseev bs-react-intl-extractor master now supports this pattern. 🎉

Can you try it and, if everything works for you, change the `message` and `translation` types to records?

Reply:
Sure! Will do tomorrow and ping you back 👍

Reply:
@alexfedoseev Did you have a chance to look into this? Alternatively, we can merge this PR and I can open a new one changing `message` and `translation` to records.

Reply:
@cknitt Sorry, I got hit by the unexpected deadline on the client project and I'll be busy until the end of this month. If you have time on this, it would be great. Otherwise, I'll be dealing with my OSS debts right after I finish with the MVP and will take care of this. Thanks!

Reply:
Ok! Will do this later today / this weekend. Will merge this PR and open a separate one.